File: 1316/c/c.kt
fun readInts(): List<Int> {
    return readLine()!!.split(" ").map(String::toInt)
}

fun main() {
    val (n, m, p) = readInts()
    val a = readInts()
    val b = readInts()
    var i = 0
    while (a[i] % p == 0)
        i++
    var j = 0
    while (b[j] % p == 0)
        j++
    println(i + j)
}

File: 1316/c/c.py
n, m, p = map(int, input().split(' '))
a = list(map(int, input().split(' ')))
b = list(map(int, input().split(' ')))
i = 0
while a[i] % p == 0:
    i += 1
j = 0
while b[j] % p == 0:
    j += 1
print(i + j)
